src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static class Constants {
+        public static final int DEFAULT_RADIUS = 10;
+        public static final String DEFAULT_COLOR = "white";
+    }
+
+    private ColorSupplier colorSupplier;
+    private Random random;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+        this.random = new Random();
+    }
+
+    public AbstractFigure getRandomFigure() {
+        int figureType = random.nextInt(5);
+
+        switch (figureType) {
+            case 0:
+                return new Square(getRandomLength(), colorSupplier.getRandomColor());
+            case 1:
+                return new Rectangle(getRandomLength(), getRandomLength(), colorSupplier
+                        .getRandomColor());
```

Comment:
```suggestion
                return new Rectangle(getRandomLength(), getRandomLength(), 
                               colorSupplier.getRandomColor());
```
a bit weird line breaking imo